### PR TITLE
Feature/6955 biography my orcid component 5

### PR DESCRIPTION
**Pull request by DanielPalafox**

### CODE REVIEW

**Comment by DanielPalafox** [comment]
[[speedup] Biography (My ORCID/Component) (5)](https://trello.com/c/3raVwrMY/6955-speedup-biography-my-orcid-component-5)

**Comment by DanielPalafox** [comment]
[[speedup] Bio: Names modal (5)](https://trello.com/c/7mmo84WT/7216-speedup-bio-names-modal-5)

File: src/app/record/components/top-bar/modals/modal-biography/modal-biography.component.scss-theme.scss [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  mat-error {
+    color: map-get($foreground, 'state-warning-dark');
+  }
+
+  mat-hint {
+    color: map-get($foreground, 'text-dark-mid');
+  }
+
```
Comment (leomendoza123) [inline src/app/record/components/top-bar/modals/modal-biography/modal-biography.component.scss-theme.scss:26]: Usea a `src/assets/scss/material.orcid.overwrites.scss/_form-field-theme.scss` to add the material overwrites here

File: src/app/record/components/top-bar/modals/modal-biography/modal-biography.component.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  ngOrcidAddYourBiography = $localize`:@@topBar.addYourBiography:Add you biography`
+
+  constructor(
+    private _platform: PlatformInfoService,
+    public dialogRef: MatDialogRef<ModalComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: UserRecord,
+    private _cdref: ChangeDetectorRef,
+    private _recordBiographyService: RecordBiographyService
+  ) {
+    this._platform.get().subscribe((platform) => {
```
Comment (leomendoza123) [inline src/app/record/components/top-bar/modals/modal-biography/modal-biography.component.ts:58]: Unsubscribe

File: src/app/record/components/top-bar/modals/modal-biography/modal-biography.component.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      visibility: visibility,
+    } as BiographyEndPoint
+  }
+
+  getBiography() {
+    this._recordBiographyService
+      .getBiography()
+      .pipe(first())
+      .subscribe((biography: BiographyEndPoint) => {
+        console.log(JSON.stringify(biography))
```
Comment (leomendoza123) [inline src/app/record/components/top-bar/modals/modal-biography/modal-biography.component.ts:108]: no logs :D

File: src/app/record/components/top-bar/modals/modal-name/modal-name.component.ts [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+          names.lastModified.day
+        )
+      } else {
+        return names.sourceName
+      }
+    }
+  }
+
+  toGivenNames() {
+    document.getElementById('given-names').scrollIntoView()
```
Comment (leomendoza123) [inline src/app/record/components/top-bar/modals/modal-name/modal-name.component.ts:276]: use the windows service to get the document. Like https://github.com/ORCID/orcid-angular/blob/129540b6d89ae718b92ab6ab0851188aeb822095/src/app/cdk/recaptcha/recaptcha.directive.ts#L110

File: src/app/record/components/top-bar/top-bar.component.ts [hunk 5]
```diff
@@ -1,12 +1,82 @@
...
+  givenNames: String = ''
+  familyName: String = ''
+  creditName: String = ''
+
+  constructor(
+    private _platform: PlatformInfoService,
+    private _user: UserService,
+    private _record: RecordService
+  ) {
+    _platform.get().subscribe((data) => {
```
Comment (leomendoza123) [inline src/app/record/components/top-bar/top-bar.component.ts:46]: unsubs

**Review by leomendoza123 — CHANGES_REQUESTED** [review]

File: src/app/record/components/top-bar/modals/modal-biography/modal-biography.component.scss-theme.scss (hunk 1 above)
Comment (DanielPalafox) [reply src/app/record/components/top-bar/modals/modal-biography/modal-biography.component.scss-theme.scss:26]: Done

File: src/app/cdk/pannel/panel/panel.component.ts [hunk 6]
```diff
@@ -1,33 +1,56 @@
...
 @Component({
   selector: 'app-panel',
   templateUrl: './panel.component.html',
   styleUrls: ['./panel.component.scss', 'panel.component.scss-theme.scss'],
 })
 export class PanelComponent implements OnInit {
-  @Input() elements: Assertion[]
   @Input() editModalComponent: ComponentType<any>
+  @Input() elements: Assertion[] | Value
+  @Input() type: 'top-bar' | 'side-bar' | 'affiliations'
```
[the remark this answers is not in the document]
Comment (DanielPalafox) [reply src/app/cdk/pannel/panel/panel.component.ts:18]: Oh my bad. I will add the missing part

**Review by leomendoza123 — APPROVED** [review]

**Pull request merged**